src/main/java/com/neovisionaries/ws/client/SocketConnector.java

```diff
@@ -107,8 +117,10 @@ private void doConnect() throws WebSocketException
 
         try
         {
-            // Connect to the server (either a proxy or a WebSocket endpoint).
-            mSocket.connect(mAddress.toInetSocketAddress(), mConnectionTimeout);
+            // Connect to the server (either a proxy or a WebSocket endpoint); overlay mSocket
+            // Ignore mConnectionTimeout
+            mSocket = mSocketFactory.createSocket(mAddress.getHostname(), mAddress.toInetSocketAddress().getPort());
+            // mSocket.connect(mAddress.toInetSocketAddress(), mConnectionTimeout);
```

Comment:
Why not add here `mSocket.setSoTimeout(mConnectionTimeout);`. It should solve the problem with timeout not being set.

Reply:
Or, even better,
``` 
mSocket = mSocketFactory.createSocket(); // This will create an unconnected socket
mSocket.setSoTimeout(mConnectionTimeout); // Set timeout
mSocket = mSocketFactory.createSocket(mSocket, mAddress.getHostname(), mAddress.toInetSocketAddress().getPort(), true); // Connect.
```

Reply:
Here is a relevant discussion https://stackoverflow.com/questions/7254389/create-sslsocket-by-sslsocketfactory-with-set-connection-timeout